.github/workflows/test/issues.test.js

```diff
@@ -0,0 +1,98 @@
+import { describe, expect, it, vi, beforeEach } from "vitest";
+import { getIssueNumber } from "../src/issues.js";
+import { createMockGithub, createMockCore } from "./mocks.js";
+
+const mockGithub = createMockGithub();
+const mockCore = createMockCore();
+
+describe("getIssueNumber", () => {
+  beforeEach(() => {
+    // Reset mocks
+    vi.clearAllMocks();
+  });
+  it("should return NaN when head_sha is missing", async () => {
+    // Call function
+    const result = await getIssueNumber({
+      head_sha: "",
+      github: mockGithub,
+      core: mockCore,
+    });
+
+    // Verify result
+    expect(result.issueNumber).toBeNaN();
+
+    // Verify appropriate message was logged
+    expect(mockCore.info).toHaveBeenCalledWith(
+      "No head_sha found in check run",
+    );
```

Comment:
I'd remove expectations about logging, it will be more work to update if we change the logs.  Logs should be flexible.

Reply:
sounds good.